File: OperationController/src/OperationController/Data/AMSConfiguration.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OperationController.Data
{
    public class AMSConfiguration
    {
        private static readonly string CONF_FILE_PATH = "../../conf";
        private static AMSConfiguration instance = new AMSConfiguration();
        public static AMSConfiguration GetInstance() { return instance; }
        public int UpdateDuration { get; set; }
        public int DetectionRadius { get; set; }
        private AMSConfiguration() 
        {
            Init();
        }

        private void Init()
        {
            using (StreamReader sr = new StreamReader(CONF_FILE_PATH))
            {
                UpdateDuration = int.Parse(sr.ReadLine());
                DetectionRadius = int.Parse(sr.ReadLine());
            }
        }
    }
}

File: AntiAirMissileSimulator/src/AntiAirMissileManager/Data/ams_configuration.cpp
#pragma once
#include "ams_configuration.h"

const std::string AMSConfiguration::CONF_FILE_PATH = "../../conf";

AMSConfiguration& AMSConfiguration::getInstance() {
    static AMSConfiguration instance;
    return instance;
}

int AMSConfiguration::getUpdateDuration() const {
    return updateDuration;
}

int AMSConfiguration::getDetectionRadius() const {
    return detectionRadius;
}

AMSConfiguration::AMSConfiguration() {
    Init();
}

void AMSConfiguration::Init() {
    std::ifstream file(CONF_FILE_PATH);

    if (file.is_open()) {
        std::string line;
        std::getline(file, line);
        updateDuration = std::stoi(line);
        std::getline(file, line);
        detectionRadius = std::stoi(line);
        file.close();
    }
    else {
        std::cerr << "Error opening configuration file: " << CONF_FILE_PATH << std::endl;
    }
}